packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -109,6 +111,13 @@ export class StatusBarImpl extends ReactWidget implements StatusBar {
         this.node.style.backgroundColor = this.backgroundColor ? this.backgroundColor : null;
     }
 
+    async setColor(color?: string): Promise<void> {
+        this.color = color;
+        this.entries.forEach(entry => {
+            entry.color = this.color ? this.color : undefined;
```

Comment:
Why `entry.color` should be updated? If a user provides a color should not we use it instead of the status bar color?

Reply:
my understanding is we want both `setBackgroundColor` for the statusbar background and `setColor` is for the font color for the entire statusbar, is there a use case where we'd have mutliple font colors that we'd like to preserve?

Reply:
The client can provide a custom color for a particular entry, then this color should be used.